app/controllers/api/v1/websites_controller.rb

```diff
@@ -4,8 +4,19 @@ def index
   end
 
   def create
-    @website = Website.create( domain: website_params[:domain], title: website_params[:title] )
-    @website.website_members.create( member_id: current_member.id, role: WebsiteMember.role.find_value(:owner).value )
+    website_exists = WebsiteMember.where("member_id = ?", current_member.id).joins(:website).where("domain = ?", website_params["domain"])
```

Comment:
asta ar trebuii sa fie o validare nu o verificare aici in controller. Astfel cand faci website.create o sa verifici daca a dat eroare si faci `_not_allowed!` nu cu verificare aici. Pentru ca se poate sa facem create si in alte parti si uitam sa adaugam verificarea asta. 

De asemenea ar trebuii sa faci `current_member.websites.create!` nu website.create si apoi websitemember.create. Incearca asa si vezi sa mearga ca sa reduci tot la o singura linie in loc de toate ifurile alea. 

Eventual sa schimbi role dupa.